### PR TITLE
Emit an error when the electron process crashes or prints to stderr

### PR DESCRIPTION
**Pull request by paulkernfeld**
Add a unit test
Allow the user to specify a path to the electron executable

### CODE REVIEW

**Comment by mappum** [comment]
Thanks, this will make the module a lot less frustrating to get working on new machines!

I looked into doing something similar at one point, but I found that there was sometimes stderr output from Electron that shouldn't trigger errors (warnings about the graphics, etc.). Are these warnings gone now? (We did upgrade to Electron v1, maybe it doesn't have warnings).

Also, I think there is an issue with the tests now, it seems to fail on Travis.

**Comment by paulkernfeld** [comment]
Good point. I don't personally get those warnings, but that's important. I redid this so that it only throws the error if the exit code is nonzero.

**Comment by paulkernfeld** [comment]
All right, looks like the tests are still failing. I think that the latest version of this PR may actually have revealed an existing problem, rather than causing a new problem.

Have you ever seen something like this? According to StackOverflow, it seems like xvfb needs to be run differently on Travis.

```
Error: Child process stderr output Xlib:  extension "RANDR" missing on display ":100". ,Xlib:  extension "RANDR" missing on display ":100".
```

**Comment by paulkernfeld** [comment]
Hmmm... not sure how to get this working. Have you ever seen this RANDR thing before?

File: src/index.js [hunk 1]
```diff
@@ -79,7 +79,8 @@ class Daemon extends EventEmitter {
     if (headless == null) {
       return cb(new Error('Could not load "headless" module'))
     }
-    headless((err, child, display) => {
+    var opts = {args: ['+extension', 'RANDR']}
```
Comment (mappum) [inline src/index.js]: This thread shows that we should be doing `-extension` to remove the RANDR extension: http://stackoverflow.com/questions/17944234/xlib-extension-randr-missing-on-display-21-trying-to-run-headless-googl

Comment (paulkernfeld) [inline src/index.js]: I tried that as well, but I got the same error.

Comment (nlocnila) [inline src/index.js]: Those arguments won't work, see: https://github.com/kesla/node-headless/blob/master/index.js#L44. It only accepts display related options.

**Comment by mappum** [comment]
Maybe we can safely ignore the RANDR error? Is it just a warning or is it actually preventing electron from running?

**Comment by mappum** [comment]
These changes were merged. 👍

**Comment by paulkernfeld** [comment]
Sorry I disappeared, and thanks for merging and fixing this!

**Pull request closed without merging**